Application Process Updates (#356)
src-d/guide: talent/application-process.md

### Visiting our careers page

- All the open roles can be found at https://sourced.tech/company/#careers where the candidate is able to apply to any desired role.

> - If there isn't a role that fits but the candidate is interested in working with us, they can send an email to talent@sourced.tech with their CV ( or GitHub account for developers) and a small intro of themselves.

Author: I agree 100%

### Screening

- Less than a week after the submission of the application the candidate receives an email with an update; however there can be delays. In any case, the candiate is free to reach out to us at any moment at talent@sourced.tech.

> - In the case of positive feedback in the application, the candidate is invited to a quick video call with the Talent Acquisition Manager.

Reviewer: ```suggestion
- If the application receives positive feedback from our initial screen, the candidate is invited to a brief video call with the Talent Acquisition Manager.
```

### Initial interview: 1st stage

> - The candidate has an introductory video call with the Talent Acquisition Manager. This is a general discussion that can last between 15-30 minutes. This is to allow us to get to know the background and experience of the candidate, as well as allowing them to get to know source{d} better. It's expected that the candidate will have already done some basic research about source{d} and has some specific questions about the role and/or the company to be expressed during the call. 

Reviewer: Is this separate from the video call described in the previous paragraph, or is it the same one?

We should probably give these names: "Initial screening call", "Hiring manager screening call", etc. (I'm not picky about the names, but having consistent terminology would help—I sometimes confuse "initial call" between the hiring manager review and the talent manager review).

Author: Yeah I totally understand what you're saying

It should flow:
1) TAM Introductory Call
2) Hiring Manager Call
3) Coding Challenge
4) Final Interview

> - The candidate has an introductory video call with the Talent Acquisition Manager. This is a general discussion that can last between 15-30 minutes. This is to allow us to get to know the background and experience of the candidate, as well as allowing them to get to know source{d} better. It's expected that the candidate will have already done some basic research about source{d} and has some specific questions about the role and/or the company to be expressed during the call. 

Reviewer: ```suggestion
- The candidate has an introductory video call with the Talent Acquisition Manager. This is a general discussion that can last between 15-30 minutes. This is to allow us to get to know the background and experience of the candidate, as well as allowing them to get to know source{d} better. It's expected that the candidate has already done some basic research about source{d} and has some specific questions about the role and/or the company to be expressed during the call. 
```

Either: "has done … and has some specific questions …" or "will have done … and have specific questions"

> - There are notable exceptions to this, such as higher volume junior engineering roles where we directly ask the developer to do the coding challenge before a call.

Reviewer: ```suggestion
- In some cases, notably for junior engineering roles where there are more candidates, we may ask the developer to do a coding challenge before a call.
```

> - In some cases, notably for junior engineering roles where there are more candidates, we may ask the developer to do a coding challenge before a call.

Reviewer 2: ```suggestion
- In some cases, notably for junior engineering roles where there are more candidates, the step below: `Coding Challenge: 3rd stage`, can happen at the beginning of the process so we may ask the developer to do the coding challenge before the call.
```

Author: sorry David I'm open to suggestions but in this case I think the example you've given is a bit waffling and complicated. Sorry!

Reviewer 2: just tried to explain that "a coding challenge" to be requested "before a call" will replace the one from 3rd step, not being an extra one; but lgtm if you think it's clear enough with the current description ;)

Author: Ah ok I see what you mean- perhaps just replace 'a' coding challenge and call with 'the'?

### Hiring Manager interview: 2nd stage

- After the initial interview, the Talent Acquisition Manager will share a summary of the call with the Hiring Manager, and will discuss the suitability of the candidate to the role.

- The Talent Acquisition Manager will contact the candidate to organise the next interview with the hiring manager, or explain why we won't be moving forward with their application

> - The Hiring Manager interview will last around one hour via video call, and will be a more indepth conversation about the candidate's experience, their role within the team, how the team works and their future tasks and responsbilities.

Reviewer: ```suggestion
- The Hiring Manager interview will last around one hour via video call, and will be a more in-depth conversation about the candidate's experience, their role within the team, how the team works, and their future tasks and responsibilities.
```

### Coding Challenge (Engineering)/ Assignment (Non-Engineering): 3rd stage 

> - For Non-engineering roles, candidates will usually be asked to complete an assignment or case study. In some cases this may be presented to members of the team in an interview.

Reviewer: ```suggestion
- For non-engineering roles, candidates will usually be asked to complete an assignment or case study. In some cases this may be presented to members of the team in an interview.
```

Either "non-Engineering" or "non-engineering"

> - Every developer who passes the first interview should complete a coding challenge remotely.

Reviewer: ```suggestion
- For engineering roles, each candidate who passes the first interview should complete a coding challenge remotely.
```

Harmonizing the grammar between the two parallel descriptions.

> - The coding challenge is currently available in Go, Python, JavaScript, Ruby, C++, Scala and Java. Other languages can be requested and we will consider the possibility of adding them. Candidates should complete the challenge in the language they feel most comfortable with.

Reviewer: ```suggestion
- Coding challenges are currently available in Go, Python, JavaScript, Ruby, C++, Scala and Java. Other languages can be requested and we will consider the possibility of adding them. Candidates should complete the challenge in the language they feel most comfortable with.
```

### Final Round of Interviews

- Non-engineering candidates who pass the third stage of the process will be invited to a final interview with various members of management which will be remote.

> - Every developer who successfully solves the coding challenge is invited to the remaining interviews to a final stage of interviews either at our offices in Madrid, or remotely.

Reviewer: ```suggestion
- Engineering candidates who successfully solve the coding challenge will be invited to a final stage of interviews either at our offices in Madrid, or remotely.
```

- Design interview (90min): to see how the developer would design a solution to a given technical problem working with several of our engineers.

> * Q&A (60min): to get to know the developer better. Two interviewers should participate; one from the team that the developer is being interviewed for, and one from any other engineering or non-engineering team (eg. Product, Talent).

Reviewer: ```suggestion
* Q&A (60min): to get to know the developer better. Two interviewers should participate; one from the team that the developer is being interviewed for, and one from any other engineering or non-engineering team (e.g., Product, Talent).
```

- Logical thinking (60min): to see how the developer solves an abstract problem in a logical way.

- In the case of rejection, the candidate will be notified in the following days after the interview. This will be a constructive discussion giving the candidate as much insight as possible. 

### Offer

> - The offer is sent by our CTO or Talent Acquisition Manager once a decision to hire has been made by the team

Reviewer: ```suggestion
- An offer is sent by our CTO or Talent Acquisition Manager once a decision to hire has been made by the team.
```

> - The candidate has no concrete time deadline to accept/reject the offer, and we are happy to have another call to discuss any questions or doubts the candidate may have. In cases where we have a limited number of open positions for a role, and a larger number of people we'd like to make an offer to, a time limit is given.

Reviewer: Should we also mention that we may also impose a deadline if we are legally required to do so?

Author: Yes this point needs discussing

Reviewer: Perhaps we can document this without making a decision on policy by writing something like:

"Except in cases where the law requires otherwise, the candidate has no fixed time deadline …"

or words to that effect? This does not commit us to a particular strategy but does clarify the possibility.

### Diversity and Remote Working

- At source{d} we believe that amazing talent should not be constrained by physical barriers, and we exhaust all avenues to hire the best candidates regardless of their location. Our multinational and multicultural team shares a culture of openness and excellence and passion for programming.

> - At the moment, we have more than 35 people from 10 different countries working closely together from our office in Madrid (Spain) and across Europe & the USA. We are more than happy to help with Visa sponsorship and guide you and your family through the whole process if you decide to come to work from our Madrid office. Currently, we have remote team members in USA, Portugal, Ireland, France, Belgium, Poland, Estonia and Russia.

Reviewer 3: and Italy 😄

- The local timezone of candidates who want to work remotely should be between San Francisco and Moscow.

> - In specific cases, such as sales roles covering a particular region, there may be a requirement 

Reviewer: I think this paragraph may be missing something.

Author: yeah I got to the end and just gave up, it should read _there may be a requirement to be based from a specific location_
